test/profiles.js

```diff
@@ -0,0 +1,104 @@
+import 'babel-polyfill'
+import { assert } from 'chai'
+import fs from 'fs'
+
+import Util from '../src/utils'
+import { Profiles, validate } from '../src/profiles'
+
+describe('Profiles', () => {
+  describe('#retrieve', () => {
+    it('returns `null` if profile ID doesn\'t exist', async () => {
+      const profiles = await new Profiles(true)
+        , retrieved = profiles.retrieve('inexistent-profile-id')
+
+      assert(retrieved === null, 'Value should be null')
+    })
+
+    it('returns remote profile by its ID', async () => {
+      const profiles = await new Profiles(true)
+        , baseProfile = fs.readFileSync('schemas/data-package.json', 'utf8')
+        , retrieved = profiles.retrieve('base')
+
+      assert.deepEqual(retrieved, JSON.parse(baseProfile))
+    })
+
+    it('returns local schema by its ID', async () => {
+      const profiles = await new Profiles(false)
+        , schema = fs.readFileSync('schemas/tabular-data-package.json',
+                                   'utf8')
+        , retrieved = profiles.retrieve('tabular')
+
+      assert.deepEqual(retrieved, JSON.parse(schema))
+    })
+  })
+
+  describe('#validate', () => {
+    it('throw array of errors if the descriptor is invalid', async () => {
+      const profiles = await new Profiles(false)
+
+      assert(profiles.validate({}) instanceof Array)
+    })
+
+    it('returns true for valid local descriptor', async () => {
+      const profiles = await new Profiles(false)
+        , datapackage = fs.readFileSync('data/dp1/datapackage.json', 'utf8')
+
+      assert(profiles.validate(datapackage))
+    })
+
+    it('returns array of lint errors for invalid json string', async () => {
+      const profiles = await new Profiles(false)
+        , descriptor = '{"test","resources":[]}'
+
+      assert(profiles.validate(descriptor) instanceof Array)
+    })
+
+    it('returns array of schema errors for invalid string descriptor',
+       async () => {
+         const profiles = await new Profiles(false)
+           , descriptor = '{"test": "shouldbename","resources":[]}'
+
+         assert(profiles.validate(descriptor) instanceof Array)
+       })
+
+    it('returns true for valid data and schema passed as argument',
+       async () => {
+         const schema = fs.readFileSync('schemas/data-package.json')
+            , profiles = await new Profiles(false)
+            , descriptor = '{"name":"test","resources":[]}'
```

Comment:
It's down in the `assert` function, I will do it outside and save the result to variable to be more readable.